clusterloader2/pkg/measurement/factory.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package measurement
+
+import (
+	"fmt"
+	"sync"
+)
+
+var factory *measurementFactory
+
+func init() {
```

Comment:
Why do you need to have global instance? Like I mentioned earlier, it makes it hard to have 2 instances of this pkg.

Reply:
I've added function that can create new instance of MeasurementFactory. Though, I'm still keeping global factory as default instance.